CMakeLists.txt

```diff
@@ -0,0 +1,345 @@
+	DEPENDS gbaGCC)
+
+
+ExternalProject_Add(LibSampleRateSource
+	URL ${LSRC_URL}
+	URL_HASH MD5=${LSRC_MD5}
+	DOWNLOAD_NO_EXTRACT TRUE
+	DOWNLOAD_DIR ${CMAKE_SOURCE_DIR}/external
+	UPDATE_COMMAND tar xzf ${LSRC_SOURCE_FOLDER}-${LSRC_VERSION}.tar.gz -C ${LSRC_SOURCE_FOLDER} --strip-components 1
+	CONFIGURE_COMMAND ${LSRC_SOURCE_FOLDER}/configure --prefix ${LSRC_INSTALL_FOLDER}
+	BUILD_COMMAND make
+	INSTALL_COMMAND make install
+	DEPENDS Tonc)
+
+set(EXTERNAL_LIBS
+	-L ${CMAKE_SOURCE_DIR}/external/tonc/code/tonclib/lib
+	 -nodefaultlibs -ltonc)
+
+set(EXTERNAL_INCLUDES
+	-I${CMAKE_SOURCE_DIR}/external/tonc/code/tonclib/include -I${CMAKE_SOURCE_DIR}/external/gcem/include)
+
+ExternalProject_Add(tools
+	SOURCE_DIR ${CMAKE_SOURCE_DIR}/tools
+	BINARY_DIR ${CMAKE_BINARY_DIR}/tools
+	BUILD_BYPRODUCTS ${CMAKE_BINARY_DIR}/tools
+	INSTALL_COMMAND cmake -E echo "Skipping install step."
+	DEPENDS LibSampleRateSource)
+
+# Export and build the resources
+build_resource("SPRITE" "tools" "${SPR_FILES}")
+build_resource("BACKGROUND" "${RESOURCES_HPP}" "${BKG_FILES}")
+build_resource("FONT" "${RESOURCES_HPP}" "${FNT_FILES}")
+build_resource("SETTING" "${RESOURCES_HPP}" "${AUS_FILES}")
+build_resource("SOUND" "${RESOURCES_HPP}" "${SND_FILES}" "${CMAKE_BINARY_DIR}/data/audio-settings.json")
+
+# Dependency target to ensure resources are exported and built
+add_custom_target(RESOURCE_OBJECTS_BUILT ALL
+	DEPENDS ${RESOURCE_OBJECT_FILES}
+	COMMENT "${RESOURCE_OBJECT_FILES}")
+	
+# Build the game files
+build_game_objects("${IWRAM_C_FILES}" "IWRAM_C")
+build_game_objects("${IWRAM_CPP_FILES}" "IWRAM_CPP")
+build_game_objects("${GAME_CPP_FILES}" "CPP")
+build_game_objects("${GAME_C_FILES}" "C")
+build_game_objects("${GAME_ASM_FILES}" "ASM")
+
+# Dependency target to ensure all the game objects are built, it implicitly depends on the resourccces being exported and built first
+add_custom_target(GAME_OBJECTS_BUILT ALL
+	DEPENDS ${GAME_IWRAM_C_OBJECT_FILES} ${GAME_IWRAM_CPP_OBJECT_FILES} ${GAME_C_OBJECT_FILES} ${GAME_CPP_OBJECT_FILES} ${GAME_ASM_OBJECT_FILES})
+
+# We need to build crt0, but not use it when building the elf, so we filter it out of all the files
+set(BUILD_GAME_FILES ${RESOURCE_OBJECT_FILES} ${GAME_IWRAM_C_OBJECT_FILES} ${GAME_IWRAM_CPP_OBJECT_FILES} ${GAME_C_OBJECT_FILES} ${GAME_CPP_OBJECT_FILES} ${GAME_ASM_OBJECT_FILES})
+list(FILTER BUILD_GAME_FILES EXCLUDE REGEX "crt0")
+
+# Build the elf
+add_custom_target(elf ALL
+	DEPENDS GAME_OBJECTS_BUILT
+	COMMAND ${GBA_CXX_COMPILER} ${LDFLAGS} -specs=${CMAKE_SOURCE_DIR}/gba.specs ${BUILD_GAME_FILES} ${EXTERNAL_LIBS} -o ${CMAKE_BINARY_DIR}/${ROM_NAME}.elf
+	BYPRODUCTS ${CMAKE_BINARY_DIR}/${ROM_NAME}.elf ${CMAKE_SOURCE_DIR}/gba.map
+	WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+
+# Dump the elf
+add_custom_target(elf_dump ALL
+	DEPENDS elf
+	COMMAND ${GBA_OBJDUMP} -dCS ${CMAKE_BINARY_DIR}/${ROM_NAME}.elf > ${CMAKE_BINARY_DIR}/${ROM_NAME}.elf.dump
+	BYPRODUCTS ${CMAKE_BINARY_DIR}/${ROM_NAME}.elf.dump)
```

Comment:
Dumping is not always required. I use it in order to analyze the compiler's output, but it could be skipped for a "final" build.

Reply:
When I add the like, check between debug and release I'll skip that in the release then. I'll add a comment to remind myself.

Reply:
Done in 748c240, the comment at least.